### ISSUE TITLE
import error in rosgraph - no netifaces module

### ISSUE THREAD

**Issue by bit-pirate** [ros/ros_comm#194]
I'm trying to get ROS compiled on ARM. After updating rosgraph to 1.9.42, I got rid of issue https://github.com/ros/ros_comm/issues/30, but didn't get much further.

My new error is:

```
$ roscore
[...]
Traceback (most recent call last):
  File "<string>", line 265, in main
  File "/opt/ros/groovy/catkin_ws/src/roslaunch/src/roslaunch/parent.py", line 268, in start
    self.runner.launch()
  File "/opt/ros/groovy/catkin_ws/src/roslaunch/src/roslaunch/launch.py", line 644, in launch
    self._setup()
  File "/opt/ros/groovy/catkin_ws/src/roslaunch/src/roslaunch/launch.py", line 620, in _setup
    self._launch_master()
  File "/opt/ros/groovy/catkin_ws/src/roslaunch/src/roslaunch/launch.py", line 384, in _launch_master
    validate_master_launch(m, self.is_core, self.is_rostest)
  File "/opt/ros/groovy/catkin_ws/src/roslaunch/src/roslaunch/launch.py", line 79, in validate_master_launch
    if not rosgraph.network.is_local_address(m.get_host()):
  File "/opt/ros/groovy/catkin_ws/src/rosgraph/src/rosgraph/network.py", line 160, in is_local_address
    local_addresses = ['localhost'] + get_local_addresses()
  File "/opt/ros/groovy/catkin_ws/src/rosgraph/src/rosgraph/network.py", line 210, in get_local_addresses
    import netifaces
ImportError: No module named netifaces
```

Digging into the code a bit, I found out that netifaces has been disabled at the top, but apparently my system is recognised as "linux-like" and hence netifaces gets imported further down.

Should I install netifaces or do sth else?

**Comment by bit-pirate** [comment]
Update: I fixed this problem for me by installing netifaces: `sudo apt-get install python-netifaces`

**Comment by dirk-thomas** [comment]
That's why the rosgraph package which uses it depends on `python-netifaces` (as of 36860c3397e29a22ee17188a674c1715e6395b18). So I guess you have just not installed all dependencies and therefore had to manually install it.

**Comment by bit-pirate** [comment]
How come that rosdep didn't pick it up?

I used `rosdep install --from-paths src --ignore-src --rosdistro groovy -y` as suggested on the wiki and it showed no missing dependency on netifaces.

**Comment by tfoote** [comment]
It should.  Can you reproduce and ticket ros/rosdep?

**Comment by bit-pirate** [comment]
Yes, I just reproduced this. python-netifaces doesn't get installed.

After looking further into this issue, I found out that the current distributed version 1.9.41 of rosgraph is missing the netifaces dependency in the package.xml. This has been fixed in 1.9.42. 

Could 1.9.42 be released soon?

**Comment by dirk-thomas** [comment]
We are waiting for the current shadow-fixed repository to get completely "green". As soon as that is the case and has been synced to public (hopefully by the end of this week) we will release the current state of the base ROS packages into shadow-fixed. These changes are than synced to public again when we are sure that there are no regressions (date not predictable - might take some time).
